Show swap error codes

lukechilds opened the pull request:
Resolves #194
Also makes a start on https://github.com/lukechilds/hyperdex/issues/194

This PR adds a new `error` property to the swap object. By default it is false so we can just do simple truthy checks on it.

If there is an error it will look like this:

```js
swap.error = {
	code: Number,
	message: String || undefined,
}
```

Currently messages are always undefined but this will change as we get more [documentation from James](https://docs.google.com/spreadsheets/d/1blgmrJCZ-SfYaJdTTVdk717kZ43R4yzm9YIW2Bt--EM/edit?usp=sharing)

I'm just displaying the code in the swap details modal but we should display a helpful message if available.

I've just added it to the modal in a basic way, feel free to change it  or style it differently.

![screen shot 2018-05-09 at 1 28 49 pm](https://user-images.githubusercontent.com/2123375/39799042-fa1b800e-538c-11e8-90b1-275af30f4fe7.png)

sindresorhus commented on app/renderer/swap-db.js:
```diff
@@ -140,6 +141,13 @@ class SwapDB {
+				// https://github.com/lukechilds/hyperdex/issues/180
+				swap.error = {
+					code: message.error,
+					message: undefined,
```
I don't think the UI should be concerned with when to show code or message. It should be handled here, so that if there's no `message`, we just use `Error Code: -231` as message. That way the consumer code doesn't need to change when the actually get messages for something things.

The below could then be changed to just:

```js
<p>
	{title(swap.statusFormatted)}
	{(swap.status === 'failed' && swap.error) && (
		<React.Fragment>
			{swap.error.message}
		</React.Fragment>
	)}
</p>
```

lukechilds replied:
Good point, will update.

lukechilds commented:
I've addresses your feedback. I also removed the `code` property so we only have message, is that what you meant?

Now I'm thinking it would be good to have the code available too.

sindresorhus commented:
> Now I'm thinking it would be good to have the code directly available too.

I think the code should still be there as a property.

lukechilds commented:
@sindresorhus fixed
